src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -337,6 +386,9 @@ void UserInfoPopup::setData(const QString &name, const ChannelPtr &channel)
     this->updateUserData();
 
     this->userStateChanged_.invoke();
+
+    this->latestMessages_->setChannel(
+        filterMessages(this->userName_, this->channel_));

```

Comment:
Should check if `latestMessages_` already holds a channel and delete the old one to not leak it since the filterMessages creates a new channel every time it is called.